app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,35 +114,73 @@ class PopupView : FrameLayout {
-        val popupDisplayFieldsWithValues: ArrayList<PopupAttribute> = ArrayList()
-        for (field in fields) {
-            popupDisplayFieldsWithValues.add(
-                PopupAttribute(
-                    field.label,
-                    popupManager.getFormattedValue(field)
-                )
-            )
-        }
-        return popupDisplayFieldsWithValues
+    private fun showSnackbarMessage(message: String) {
+        Snackbar.make(this, message, Snackbar.LENGTH_LONG).show()
     }
 
     /**
-     * Represents an object comprising of Popup display field and its value.
+     * Saves the popup by applying changes to the feature service.
      */
-    data class PopupAttribute(val field: String, val value: String)
+    private fun savePopup() {
+        val editingErrorFuture: ListenableFuture<ArcGISRuntimeException> = popupManager.finishEditingAsync()
+        editingErrorFuture.addDoneListener {
+            try {
+                editingErrorFuture.get()
+
+                val feature: Feature = popupManager.popup.geoElement as Feature
+                val featureTable: FeatureTable = feature.featureTable
+                if (featureTable is ServiceFeatureTable) {
+                    val applyEditsFuture: ListenableFuture<List<FeatureEditResult>> =
+                        featureTable.applyEditsAsync()
+                    applyEditsFuture.addDoneListener {
+                        try {
+                            val featureEditResults: List<FeatureEditResult> = applyEditsFuture.get()
+                            // Check for errors in FeatureEditResults
+                            if (featureEditResults.any { result -> result.hasCompletedWithErrors() }) {
+                                showSnackbarMessage("There was a problem saving the feature")
+                            } else {
+                                showSnackbarMessage("Feature saved successfully")
+                            }
+
+                        } catch (ie: InterruptedException) {
+                            Logger.e(
+                                "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                                    ie
+                                )
+                            )
+                        } catch (ee: ExecutionException) {
+                            Logger.e(
+                                "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                                    ee
+                                )
+                            )
+                        }
+                    }
+                }
+            } catch (ie: InterruptedException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ie
+                    )
+                )
+            } catch (ee: ExecutionException) {
```

Comment:
No need for two catch blocks. Just catch `Exception` and log message. I wonder if this needs more user feedback other than just a log.

Reply:
I have removed two catch blocks. 
In case of no network on the client side. An exception is thrown when the app is not able to apply the edits. I am showing the message of the exception in a dialog box to the user.